__tests__/topological_sort_test.re

```diff
@@ -0,0 +1,57 @@
+open Jest;
+open Expect;
+
+describe("Topological Sort", () => {
+    open TopologicalSort;
+
+    test("single node", () => {
+        let adj_list = [{id: "A", children: []}];
+        let expected_sorting = ["A"];
+        let sorting = sort(adj_list);
+
+        expect(sorting) |> toEqual(expected_sorting);
+    });
+
+    test("simple tree", () => {
+
+        let adj_list = [
+            {id: "A", children: ["B"]},
+            {id: "B", children: ["C"]}, 
+            {id: "C", children: ["D"]}, 
+            {id: "D", children: []}
+            ];
+
+            let expected_sorting = ["A", "B", "C", "D"];
+            let sorting = sort(adj_list);
+    
+            expect(sorting) |> toEqual(expected_sorting);
+    });
+    
+    test("star graph", () => {
+
+        let adj_list = [
+            {id: "A", children: []},
+            {id: "B", children: ["A"]}, 
+            {id: "C", children: ["A"]}, 
+            {id: "D", children: ["A"]}];
+
+            let expected_sorting = ["D", "C", "B", "A"];
+            let sorting = sort(adj_list);
+    
+            expect(sorting) |> toEqual(expected_sorting);
+    });
+
+    test("forest", () => {
```

Comment:
Can we add a test case similar to lecture 14 page 5? And make sure to visit `B` first?